src/main/java/software/amazon/awssdk/crt/http/HttpClientConnectionManagerOptions.java

```diff
@@ -83,6 +85,23 @@ public HttpClientConnectionManagerOptions withTlsContext(TlsContext tlsContext)
      */
     public TlsContext getTlsContext() { return tlsContext; }
 
+    /**
+     * Sets the connection-specific TLS options to use for connections in the connection pool.
+     * Either TLS context or TLS connection options will be enough to set up TLS connection.
+     * If both set, TLS connection options will override the TLS context was set.
+     * @param tlsConnectionOptions The TlsConnectionOptions to use
+     * @return this
+     */
+    public HttpClientConnectionManagerOptions withTlsConnectionOptions(TlsConnectionOptions tlsConnectionOptions) {
```

Comment:
instead of the tlsConnectionOptions silently overriding the tlsContext, I vote that we make it an error to set both

TLS stuff seems like a scary category of stuff to silently override